### ISSUE TITLE
[DOCS]: github_repository_collaborator

### ISSUE THREAD

**Issue by sneakyjoeru** [integrations/terraform-provider-github#2075]
### Describe the need

Collaborator fights with teams over access level similarly to collaborators.
I specifically used collaborator as it was indicated that it will not fight over rights in a way collaborators do, but it seems to not be the case. When I configured teams and multiple instances of collaborator, if those collaborator records also are in teams, terraform tries to switch their access to the one specified in a team.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by kfcampbell** [comment]
@sneakyjoeru do you mind being a bit more specific and providing a minimal reproductive case? I need a little more information to troubleshoot. Is the behavior dependent on the order the resources were applied in?

**Comment by sneakyjoeru** [comment]
Sure. 
So, you have notes in [collaborators](https://registry.terraform.io/providers/integrations/github/latest/docs/resources/repository_collaborators) doc about it conflicting with collaborator and teams.
There's no such note for conflicting with collaborators in [collaborator](https://registry.terraform.io/providers/integrations/github/latest/docs/resources/repository_collaborator) page.
Still, I have several [team_repository](https://registry.terraform.io/providers/integrations/github/latest/docs/resources/team_repository) resources which are said to conflict only with [collaborators](https://registry.terraform.io/providers/integrations/github/latest/docs/resources/repository_collaborators) , but when I try to add a collaborator with higher rights than it's declared in team_repository, it tries to bring permissions of those collaborator records to match what's described in team_repository.

TLDR trying to elevate permissions for "collaborator" which is already listed in "team_repository" with lower level.

**Comment by kfcampbell** [comment]
@sneakyjoeru No worries, thank you for caring enough to open an issue and talk through it.
